hyperspy/axes.py

```diff
@@ -515,11 +552,33 @@ def value2index(self, value, rounding=round):
         ------
         ValueError
             If any value is out of the axis limits.
+            If the value is incorrectly formatted.
 
         """
         if value is None:
             return None
 
+        if isinstance(value, str):
+            if value == "":
+                raise ValueError("Cannot slice with an empty string")
+            # if first character is a digit, try unit conversion
+            # otherwise try relative (fractional) indexing
+            if value[0].isdigit():
+                try:
+                    value = self._get_value_from_value_with_units(value)
+                except:
+                    raise ValueError(f"Expected calibrated value with units like '20nm' but '{value}' could not be interpreted this way.")
+            else:
+                try:
+                    assert value.startswith('rel')
+                    relative_value = float(value[3:])
+                except (AssertionError, ValueError):
+                    raise ValueError(f"Expected relative units of kind 'rel0.2' but '{value}' could not be interpreted this way.")
```

Comment:
Similar as above, I am not convinced this is necessary and it would be better to use `_get_value_from_relative_string` to avoid repetition.